### PR TITLE
Create Filter API to get Cases by Date or range of date.

### LINKED ISSUE #205: Create Filter API to get Cases by Date or range of date.
**Is your feature request related to a problem? Please describe.**
Filtering Cases by Date or range of date.
Create API to fetch cases by date or range of date. (this will be useful in frontend analytics)

### LINKED ISSUE #208: Bug in API for getting list of cases
**Describe the bug**
There is a bug in API: http://127.0.0.1:5000/case/list
```
@case.route('/list', methods=["GET"])
 def list():
    all_cases = Case.query.order_by(Case.timestamp).all()
    result = cases_schema.dump(all_cases)
    return jsonify(result)
```
Case model does not have timestamp column.

**To Reproduce**
We can use extracted_on instead.

### PR DESCRIPTION
**Pull request by swapnalshahil**
# Description

Filtering Cases by Date or range of date.
Create API to fetch cases by date or range of date. (this will be useful in frontend analytics)

Fixes #205 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

POSTMAN

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by swapnalshahil** [comment]
Some of the minor changes are due to the linter which deletes unnecessary space there.

**Comment by swapnalshahil** [comment]
@shivanshu1333 sir! Please review.

File: flask-backend/api/routes/case.py [hunk 1]
```diff
@@ -117,6 +118,53 @@ def openFile(case_name, folder_name, file_name):
...
+        to_date = str(req['end date'])
+
+        if not to_date:
+            '''
+            if end date is not there
+            then end date will be the present date.
+            '''
+            to_date = str(datetime.date.today())
+
+    except:
```
Comment (shivanshuraj1333) [inline flask-backend/api/routes/case.py]: instead through an Exception here like:
    except Exception as e:
        response = {
            "success": False,
            "message": "please provide date."
        }
        return make_response(jsonify(response)), error code

File: flask-backend/api/routes/case.py [hunk 2]
```diff
@@ -117,6 +118,53 @@ def openFile(case_name, folder_name, file_name):
...
+        time = time[0]
+
+        if time >= from_date and time <= to_date:
+            case_filtered.append((cases.data_path, cases.extracted_on))
+
+    if case_filtered:
+
+        return jsonify(case_filtered)
+    else:
+        return "No case found in this range", 404
```
Comment (shivanshuraj1333) [inline flask-backend/api/routes/case.py]: same as above

**Review by shivanshuraj1333 — CHANGES_REQUESTED** [review]

File: flask-backend/api/routes/case.py [hunk 3]
```diff
@@ -117,6 +118,53 @@ def openFile(case_name, folder_name, file_name):
...
+        If starting date is provided and no end date then 
+        cases from starting date till present date will be shown.
+    '''
+    try:
+        req = request.get_json()
+
+        from_date = str(req['starting date'])
+
+        to_date = str(req['end date'])
+
```
Comment (sachinsom93) [inline flask-backend/api/routes/case.py:137]: @swapnalshahil you can add an additional check for equal vales of to_date from from_date :)

File: flask-backend/api/routes/case.py (hunk 2 above)
Comment (swapnalshahil) [reply flask-backend/api/routes/case.py]: Done!

File: flask-backend/api/routes/case.py (hunk 1 above)
Comment (swapnalshahil) [reply flask-backend/api/routes/case.py]: @shivanshu1333  Made changes sir!

File: flask-backend/api/routes/case.py (hunk 3 above)
Comment (swapnalshahil) [reply flask-backend/api/routes/case.py:137]: Hey for getting cases of only a particular date we have to write the same date on "starting date" as well as on the "end date". I thought about it earlier but I will use this filter in the analytics section to filter cases within range.
Just like you saw on any job filter portal or on Flipkart filters.

Comment (sachinsom93) [reply flask-backend/api/routes/case.py:137]: Cool,  got your point 👍🏼

**Review by shivanshuraj1333 — APPROVED** [review]

**Pull request merged**